velox/dwio/common/DataSink.h

```diff
@@ -195,6 +196,43 @@ class MemorySink : public DataSink {
   DataBuffer<char> data_;
 };
 
+class WriteFileSink : public DataSink {
```

Comment:
What's the purpose of this class? How is it used?

Reply:
It is used as a data sink to write to hdfs file or local file. Sink to hdfs is not supported in the FileSink implemention，so I implement this one to support.

Reply:
Got it. Thank you for clarifying. Let's add an e2e test to run a query that reads and writes from HDFS to make sure all these code paths are exercised.